src/components/PaginationButtons.js

```diff
@@ -0,0 +1,59 @@
+import '../styles/pageButton.css';
+
+function PageButton({ children, onClick, isActive, isLoading }) {
+  return (
+    <button
+      onClick={onClick}
+      className={`page-button ${isActive ? 'page-button__active' : ''}`}
+      disabled={isLoading}
+    >
+      {children}
+    </button>
+  );
```

Comment:
```suggestion
  const handleClick = () => {
    if (!isLoading) {
      onClick();
    }
  };

  return (
    <button
      onClick={handleClick}
      className={`page-button ${isActive ? 'page-button__active' : ''}`}
    >
      {children}
    </button>
  );
```
불필요한 통신 요청을 막는다는 의도는 아주 좋습니다 👍
다만, 그로인해 화면에 눈에 띄게 깜빡임 현상이 나오고 있으므로 이런식으로 처리를 해주시는 게 좋을 것 같습니다.